dotnet/src/Connectors/Connectors.Google/Core/Gemini/Clients/GeminiChatCompletionClient.cs

```diff
@@ -517,61 +516,17 @@ private static bool CheckAutoInvokeCondition(Kernel? kernel, GeminiPromptExecuti
         return autoInvoke;
     }
 
-    private static void ValidateAndPrepareChatHistory(ChatHistory chatHistory)
+    private static void ValidateChatHistory(ChatHistory chatHistory)
     {
         Verify.NotNullOrEmpty(chatHistory);
-
-        if (chatHistory.Where(message => message.Role == AuthorRole.System).ToList() is { Count: > 0 } systemMessages)
-        {
-            if (chatHistory.Count == systemMessages.Count)
-            {
-                throw new InvalidOperationException("Chat history can't contain only system messages.");
-            }
-
-            if (systemMessages.Count > 1)
-            {
-                throw new InvalidOperationException("Chat history can't contain more than one system message. " +
-                                                    "Only the first system message will be processed but will be converted to the user message before sending to the Gemini api.");
-            }
-
-            ConvertSystemMessageToUserMessageInChatHistory(chatHistory, systemMessages[0]);
-        }
-
-        ValidateChatHistoryMessagesOrder(chatHistory);
+        ThrowIfChatContainsOnlySystemMessages(chatHistory);
     }
 
-    private static void ConvertSystemMessageToUserMessageInChatHistory(ChatHistory chatHistory, ChatMessageContent systemMessage)
+    private static void ThrowIfChatContainsOnlySystemMessages(ChatHistory chatHistory)
```

Comment:
nit: Do we need a separate method for this check? If it's used only in one place, we can put this logic in `ValidateChatHistory` directly.